test/test_llogcolor.py

```diff
@@ -0,0 +1,101 @@
+"""A test suite for comparing the text output
+of llogcolor for an assumed-to-be-correct output.
+
+The pager is skipped for these tests, and the
+color control codes go strait to a file, so they get saved.
+"""
+
+import filecmp
+import pathlib
+import sys
+import tempfile
+import unittest
+
+# now I need to import llogcolor
+# this isn't the normal way to import,
+# but there are 2 issues.
+# 1. this is not a python only project, I don't want
+# to structure all of lustre-tools-llnl as a python package
+# 2. the file to be tested doesn't have '.py' suffix
+# from https://stackoverflow.com/questions/2601047/
+from importlib.util import spec_from_loader, module_from_spec
+from importlib.machinery import SourceFileLoader
+
+path_to_llogcolor = str(pathlib.Path("../scripts/llogcolor").resolve())
+spec = spec_from_loader(
+    "llogcolor", SourceFileLoader("llogcolor", path_to_llogcolor)
+)
+llogcolor = module_from_spec(spec)
+spec.loader.exec_module(llogcolor)
+
+#### paths to input logs ###
+# small and simple log file, all of the same thread ID
+in_simple = "llogcolor_files/sample_logs"
```

Comment:
It would be good for the filename to reflect what is special/interesting about the file, e.g. raw.single_thread

Reply:
And, it might be useful to make it clear from the filenames which are input and which are (expected) output.

Reply:
Yes, I'll give them better names.